go.mod

```diff
@@ -16,7 +16,7 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/google/uuid v1.3.0
 	github.com/hashicorp/go-azure-helpers v0.57.0
-	github.com/hashicorp/go-azure-sdk v0.20230623.1103505
+	github.com/hashicorp/go-azure-sdk v0.20230704.1182735
```

Comment:
`go-azure-sdk` dependency updates should be done in their own PR so as not to conflict with other changes, would you mind splitting this one out?